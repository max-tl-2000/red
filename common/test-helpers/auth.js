/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import moment from 'moment'; // eslint-disable-line red/no-moment
import { expect } from 'chai';

export const verifyTokenExpiration = (decodedToken, expectedTokenExpiration) => {
  const iat = moment.unix(decodedToken.iat);
  const exp = moment.unix(decodedToken.exp);

  const result = exp.diff(iat, expectedTokenExpiration.substring(expectedTokenExpiration.length - 1));
  const expectedValue = expectedTokenExpiration.substring(0, expectedTokenExpiration.length - 1);
  expect(result).to.equal(+expectedValue);
};
