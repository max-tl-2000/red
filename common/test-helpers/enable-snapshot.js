/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import chai from 'chai';
import chaiJestSnapshot from 'chai-jest-snapshot';

const enableSnapshot = () => {
  if (!global.before) return;
  chai.use(chaiJestSnapshot);
  before(() => {
    chaiJestSnapshot.resetSnapshotRegistry();
  });

  beforeEach(function beforeEach() {
    const { currentTest } = this;
    chaiJestSnapshot.setFilename(`${currentTest.file}.chai-snap`);
    chaiJestSnapshot.setTestName(currentTest.fullTitle());
  });
};

enableSnapshot();

export { editUnpredictable } from '../helpers/snapshot-helper';
