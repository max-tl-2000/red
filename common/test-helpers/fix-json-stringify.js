/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

const stringify = JSON.stringify;

before(() => {
  JSON.stringify = obj => {
    const seen = [];

    return stringify(obj, (key, val) => {
      if (typeof val === 'object') {
        if (seen.indexOf(val) >= 0) {
          return;
        }
        seen.push(val);
      }
      return val; // eslint-disable-line
    });
  };
});

after(() => {
  JSON.stringify = stringify;
});
