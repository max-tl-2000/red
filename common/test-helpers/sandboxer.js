/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

let _sinon;
let sandboxes = [];

function create() {
  // this is for when this module is used in karma
  _sinon = _sinon || global.sinon;
  const sandbox = _sinon.sandbox.create();
  sandboxes.push(sandbox);

  return sandbox;
}

function restore() {
  sandboxes.forEach(sandbox => sandbox.restore());
  sandboxes = [];
}

function setSinon(sinon) {
  _sinon = sinon;
}

export { create, restore, setSinon };
