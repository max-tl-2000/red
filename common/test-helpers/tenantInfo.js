/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

export const TEST_TENANT_ID = 'af20764d-1e0d-41a5-be54-3766215f89a4';
