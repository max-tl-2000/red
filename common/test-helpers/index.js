/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
import { unmountComponentAtNode } from 'react-dom';
import when from './when';
import and from './and';
import * as overrider from './overrider';
import * as sandboxer from './sandboxer';
import sleep from '../helpers/sleep';

sandboxer.setSinon(sinon);

const expect = chai.expect;
const assert = chai.assert;

chai.use(sinonChai);

let div;

if (global.document) {
  beforeEach(() => {
    div = document.createElement('div');
    div.setAttribute('id', 'sandbox');

    document.body.appendChild(div);
  });
}

afterEach(() => {
  if (global.document && div) {
    unmountComponentAtNode(div);
    document.body.removeChild(div);
    div = null;
  }

  overrider.restore();
  sandboxer.restore();
});

const getSandBoxDiv = () => div;

// helper method to trigger a real click event in the browser
// jquery.fn.click is not the same as this as the event need
// to also generate the capture phase
const triggerMouseEvent = (ele, evt) => {
  const mEvent = document.createEvent('MouseEvent');
  // interface
  // event.initMouseEvent(type, canBubble, cancelable, view,
  //                    detail, screenX, screenY, clientX, clientY,
  //                    ctrlKey, altKey, shiftKey, metaKey,
  //                    button, relatedTarget);
  mEvent.initMouseEvent(evt, true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);

  ele.dispatchEvent(mEvent);
};

export { getSandBoxDiv, triggerMouseEvent, overrider, expect, assert, sinon, when, sandboxer, and, sleep };
