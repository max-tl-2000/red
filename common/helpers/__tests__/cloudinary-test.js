/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import * as overrider from '../../test-helpers/overrider';
/* eslint-disable global-require */

describe('cloudinary', () => {
  let cloudinary;
  let ov;
  let globals;

  beforeEach(() => {
    jest.resetModules();
    globals = require('../globals');
    ov = overrider.create(globals);
  });

  afterEach(() => {
    overrider.restore();
  });

  describe('getUrlFromCloudinary', () => {
    it('should get same url back if an attempt to use the module is done without calling init', () => {
      cloudinary = require('../cloudinary');

      const url = cloudinary.getUrlFromCloudinary('http://handle/path/to/image.png');
      expect(url).toEqual('http://handle/path/to/image.png');
    });

    it('should return a wrapped url if cloudinary cloudName is set', () => {
      cloudinary = require('../cloudinary');
      cloudinary.init({ cloudName: 'roy' });
      const url = cloudinary.getUrlFromCloudinary('http://some/path/to/image.jpg');
      expect(url).toMatchSnapshot();
    });

    it('should handle passing optional options as well', () => {
      cloudinary = require('../cloudinary');
      cloudinary.init({ cloudName: 'roy' });
      const url = cloudinary.getUrlFromCloudinary('http://some/path/to/image.jpg', ['w_200', 'h_400']);
      expect(url).toMatchSnapshot();
    });

    it('should return a wrapped url when not running from a cucumber like environmet', () => {
      ov.override('location', {
        hostname: 'some.env.reva.tech',
      });

      cloudinary = require('../cloudinary');

      cloudinary.init({ cloudName: 'testCloud' });

      const url = cloudinary.getUrlFromCloudinary('http://handle/path/to/image.png');
      expect(url).toMatchSnapshot();
    });

    it('should return the original url when running from a cucumber like environmet', () => {
      ov.override('location', {
        hostname: 'cucumber-1.env.reva.tech',
      });

      cloudinary = require('../cloudinary');

      cloudinary.init({ cloudName: 'testCloud' });

      const url = cloudinary.getUrlFromCloudinary('http://handle/path/to/image.png');
      expect(url).toEqual('http://handle/path/to/image.png');
    });
  });

  describe('getBigAvatar', () => {
    it('should create the url to retrieve big avatars', () => {
      cloudinary = require('../cloudinary');
      cloudinary.init({ cloudName: 'roy' });
      const url = cloudinary.getBigAvatar('http://some/path/to/image.jpg');
      expect(url).toMatchSnapshot();
    });
  });

  describe('getSmallAvatar', () => {
    it('should create the url to retrieve small avatars', () => {
      cloudinary = require('../cloudinary');
      cloudinary.init({ cloudName: 'roy' });
      const url = cloudinary.getSmallAvatar('http://some/path/to/image.jpg');
      expect(url).toMatchSnapshot();
    });
  });

  describe('getImageForEmail', () => {
    it('should create the url to retrieve images for email templates', () => {
      cloudinary = require('../cloudinary');
      cloudinary.init({ cloudName: 'test' });
      const url = cloudinary.getImageForEmail('http://some/path/to/image.jpg', [
        { key: 'w', value: 2 },
        { key: 'h', value: 3 },
        { key: 'unkown', value: 'somevalue' },
      ]);
      expect(url).toContain('w_2,h_3');
      expect(url).not.toContain('unkown');
    });
  });
});
