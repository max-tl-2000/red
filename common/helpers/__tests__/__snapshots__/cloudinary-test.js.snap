// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cloudinary getBigAvatar should create the url to retrieve big avatars 1`] = `"https://res.cloudinary.com/roy/image/fetch/w_64,z_0.8,ar_1:1,c_thumb,g_face:center,f_auto,q_auto:good,r_max,e_auto_brightness,fl_force_strip/http%3A%2F%2Fsome%2Fpath%2Fto%2Fimage.jpg"`;

exports[`cloudinary getSmallAvatar should create the url to retrieve small avatars 1`] = `"https://res.cloudinary.com/roy/image/fetch/ar_1:1,c_thumb,g_face:center,f_auto,q_auto:good,r_max,e_auto_brightness,fl_force_strip,cs_no_cmyk,dpr_1.0,w_40,z_0.9/l_fetch:,fl_cutter/http%3A%2F%2Fsome%2Fpath%2Fto%2Fimage.jpg"`;

exports[`cloudinary getUrlFromCloudinary should handle passing optional options as well 1`] = `"https://res.cloudinary.com/roy/image/fetch/w_200,h_400/http%3A%2F%2Fsome%2Fpath%2Fto%2Fimage.jpg"`;

exports[`cloudinary getUrlFromCloudinary should return a wrapped url if cloudinary cloudName is set 1`] = `"https://res.cloudinary.com/roy/image/fetch/http%3A%2F%2Fsome%2Fpath%2Fto%2Fimage.jpg"`;

exports[`cloudinary getUrlFromCloudinary should return a wrapped url when not running from a cucumber like environmet 1`] = `"https://res.cloudinary.com/testCloud/image/fetch/http%3A%2F%2Fhandle%2Fpath%2Fto%2Fimage.png"`;
